code/functions.jl

```diff
@@ -0,0 +1,166 @@
+"""
+TODO
+"""
+function generate(x, σ)
+    @assert σ >= 0.0
+    return σ > 0.0 ? rand(TruncatedNormal(x,σ, 0.0, 100.0)) : x
+end
+
+"""
+***Simulations for the host population and specialist parasitoid population dynamics for the first 50 generations***
+
+- `N`: Initial host population size
+- `P`: Initial parasitoid population size
+- `t`: Number of generations
+- `f`: Type of parasite chosen to to the simulations (specialist or generalist)
+- `F`: Finite rate of increase of the host population
+- `D`: Density independent mortality (as a probability of survival)
+- `c`: Number of parasitoids emerging from each host parasitized
+- `a`: Searching efficiency (per capita)
+- `h`: Saturation number of parasitoids
+- `b`: Rate of appoaching h
+- `th`: Handling time (as a proportion of the total time)
+- `m`: Extent of clumping of the parasitoid attacks
+
+Return : `dynamics`: Matrix containing the host and parasite density for the initial populations and for every generation simulated
+"""
+function simulation(N::Float64, P::Float64; t::Int64=50, f=specialist_dyn, F=4.0, D=0.5, c=1.0, a=0.5, h=10.0, b=25.0, th=0.0, m=0.2, D_sd=0.0, a_sd=0.0, h_sd=0.0, c_sd=0.0)
```

Comment:
I would split to keyword arguments at `F`. A long list of unnamed parameters can be hard to parse.